### ISSUE TITLE
Ability to anchor to headers

### ISSUE THREAD

**Issue by psychobunny** [julianlam/nodebb-plugin-markdown#80]
http://stackoverflow.com/questions/6695439/how-to-link-to-a-named-anchor-in-multimarkdown

So IMO everytime somebody does something like 

`## This is a header`

It should generate a link such as this, using something similar to slugify:

`<h2><a name="this-is-a-header"></a>This is a header`

This way people can create for example a table of contents on a long post.

**Comment by pitaj** [comment]
Just gonna leave this here:
https://github.com/jonschlinkert/markdown-toc

Oh, and commonmark is cool, too:
https://www.npmjs.com/package/commonmark

**Comment by abhinov2** [comment]
How and where to add fa fa icon in markdown header.
there is fa fa icon of $ through which inline math is coming on composer , where it has been done ,
Guide me through.

**Comment by julianlam** [comment]
@abhinov2 No. Stop posting comments to random issues. Use the forum instead: https://community.nodebb.org

**Comment by julianlam** [comment]
@psychobunny @pitaj Header anchors were removed from markdown-it for... reasons. I can't really recall right now.

Jon Schlinker's plugin does support it but markdown-it is a hard fork away, so we stick with it.

It is available as a plugin however: https://www.npmjs.com/package/markdown-it-anchor, potentially could add it to the markdown plugin.
